ecmascript/transforms/typescript/src/strip.rs

```diff
@@ -2217,6 +2217,34 @@ where
         d.visit_mut_children_with(self);
         d.definite = false;
     }
+
+    fn visit_mut_pat_or_expr(&mut self, node: &mut PatOrExpr) {
+        // Coerce bindingident to assign expr where parenthesis exists due to TsAsExpr
+        // like (warn as any) = v;
+        // We want to do this _before_ visiting children, otherwise handle_expr
+        // wipes out TsAsExpr and there's no way to distinguish between
+        // plain (x) = y vs. (x as any) = y;
+        match node {
+            PatOrExpr::Pat(pat) => match &**pat {
+                Pat::Expr(expr) => match &**expr {
+                    Expr::Paren(ParenExpr { expr, .. }) => match &**expr {
+                        Expr::TsAs(TsAsExpr { expr, .. }) => match &**expr {
+                            Expr::Ident(ident) => {
+                                *node = PatOrExpr::Pat(Box::new(Pat::Ident(ident.clone().into())));
+                            }
+                            _ => (),
+                        },
+                        _ => (),
+                    },
+                    _ => (),
+                },
+                _ => (),
+            },
+            _ => (),
+        }
+
+        node.visit_mut_children_with(self);
```

Comment:
I think checking for `Expr::Ident` after `node.visit_mut_children_with(self);` would be more clear and hande more cases.

Reply:
Do you mean before `match` to traverse node to find out ParenExpr? I tried it and hit by this https://github.com/swc-project/swc/pull/2631/files#diff-aee7604a9d35c517e7e1637b34d66b7ce8de17cc407ba6236d314471d00d4b5eR2224-R2226 - which invalidates few existing test cases like `(x) = v`. It's different to TsAsExpr which strips paren (which is expected) need to preserve as-is.

Reply:
visit_mut_children triggers to visit expr, and strips out TsAsExpr with `handle_expr` so there wasn't way to distinguish if it's coming from as-casting vs. ecmascript runtime paren.

Reply:
Oh, I see, then it's fine.